crates/p4-analyzer/src/native_fs.rs

```diff
@@ -0,0 +1,173 @@
+pub mod native_fs {
+	use analyzer_abstractions::{
+		fs::{AnyEnumerableFileSystem, EnumerableFileSystem},
+		lsp_types::{TextDocumentIdentifier, Url},
+		BoxFuture,
+	};
+	use glob::glob;
+
+	pub struct NativeFs {}
+
+	impl NativeFs {
+		pub fn new() -> AnyEnumerableFileSystem { Box::new(NativeFs {}) }
+	}
+
+	// EnumerableFileSystem part of NativeFs will just use std::fs methods for the functions
+	impl EnumerableFileSystem for NativeFs {
+		fn enumerate_folder<'a>(
+			&'a self,
+			folder_uri: Url,
+			file_pattern: String,
+		) -> BoxFuture<'a, Vec<TextDocumentIdentifier>> {
+			//self.start_folder_watch(&folder_uri); // add folder to watch list
+
+			async fn enumerate_folder(folder_uri: Url, file_pattern: String) -> Vec<TextDocumentIdentifier> {
+				let folder = folder_uri.to_file_path().unwrap();
+				let glob_pattern = folder.to_str().unwrap().to_owned() + file_pattern.as_str();
+
+				let mut output = Vec::new();
+
+				for files in glob(glob_pattern.as_str()).expect("Failed to read glob pattern") {
```

Comment:
This is currently panicking when run from VSCode:

```
thread 'main' panicked at 'Failed to read glob pattern: PatternError { pos: 34, msg: "recursive wildcards must form a single path component" }', crates/p4-analyzer/src/native_fs.rs:30:58
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I suspect it's because the URI provided by VSCode doesn't end with a `'/'` character, and when concatenated with the `file_pattern` produces an invalid glob pattern (i.e., it'll be something like 'dirA/dirB**.p4').